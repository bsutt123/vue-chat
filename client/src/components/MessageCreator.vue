<template>
  <div :class="$style.container">
    <form v-on:submit.prevent="submit" :class="$style.form">
      <textarea
        type='text'
        v-model="newMessage"
        :class="$style.newMessageInput"
      >
      </textarea>
      <button
        type="submit"
        :class='$style.submitButton'
        class='btn'
      >
        Submit
      </button>
    </form>
  </div>
</template>

<script>
import { createNamespacedHelpers } from 'vuex';

const { mapActions } = createNamespacedHelpers('messages');
export default {
  data() {
    return {
      newMessage: '',
    };
  },
  methods: {
    ...mapActions([
      'createMessage',
    ]),
    submit() {
      this.createMessage(this.newMessage);
      this.newMessage='';
    },
  },
};
</script>

<style lang="postcss" module>
.container {
  position: fixed;
  background: white;
  left: 15rem;
  bottom: 0;
  right: 0;
}

.form {
  display: grid;
  grid-template-columns: 1fr auto;
  grid-template-rows: 4rem;
  grid-gap: .5rem;
}

.submitButton {
  align-self: center;
  margin: 0;
  font-size: 1.25rem;
}

.newMessageInput {
  font-size: 1rem;
}
</style>
