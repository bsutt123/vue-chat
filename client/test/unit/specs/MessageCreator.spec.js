import { shallow, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import VModal from 'vue-js-modal';

import { CssModuleTestHelperMixin } from '../utils/test-helper';
import storeObject from '../utils/testStore';
import MessageCreator from '@/components/MessageCreator';

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(VModal);
localVue.use(CssModuleTestHelperMixin);


describe('MessageCreator', () => {
  let store;
  let actions;
  let wrapper;

  beforeEach( () => {
    store = new Vuex.Store(storeObject);
    actions = storeObject.modules.messages.actions;
    wrapper = shallow(MessageCreator, { localVue, store })
    wrapper.set
  });

  it('renders the snapshot correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  })

  it('creates a message on submit', () => {
    const form = wrapper.find('.form');
    form.trigger('submit');
    expect(actions.createMessage.mock.calls.length).toEqual(1);
  })

  it('renders newMessage', () => {
    const input = wrapper.find('.newMessageInput');
    wrapper.setData({newMessage: 'Brady'});
    expect(wrapper.vm.$data.newMessage).toBe('Brady');
  })
})