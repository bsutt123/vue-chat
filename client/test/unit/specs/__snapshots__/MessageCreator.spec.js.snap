// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageCreator renders the snapshot correctly 1`] = `
<div class="container">
    <form class="form">
        <textarea type="text" class="newMessageInput"></textarea>
        <button type="submit" class="btn submitButton">
            Submit
        </button>
    </form>
</div>
`;
